### PR TITLE
docs: add documentation on debtor groups

### PR DESCRIPTION
**Pull request by jniles**
Adds documentation on debtor groups and the different options available to administrators.  This is an iterative process.

-----
[View rendered docs/en/finance/bookkeeping/patient-invoices.md](https://github.com/jniles/bhima/blob/docs-debtor-groups-and-conventions/docs/en/finance/bookkeeping/patient-invoices.md)
[View rendered docs/en/finance/core-concepts/debtor-groups.md](https://github.com/jniles/bhima/blob/docs-debtor-groups-and-conventions/docs/en/finance/core-concepts/debtor-groups.md)
[View rendered docs/en/finance/core-concepts/index.md](https://github.com/jniles/bhima/blob/docs-debtor-groups-and-conventions/docs/en/finance/core-concepts/index.md)
[View rendered docs/en/finance/index.md](https://github.com/jniles/bhima/blob/docs-debtor-groups-and-conventions/docs/en/finance/index.md)

### CODE REVIEW

File: docs/en/finance/bookkeeping/patient-invoices.md [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 BHIMA packages a powerful, flexible, and simple patient invoicing module.  Patients are invoiced for goods and services in the [inventory](#).  At the time of invoicing, the price of individual items are calculated from the base price of the inventory and any price lists that may apply to the patient.  Furthermore, an additional charge, known as an [Invoice Fee](#), or an additional reduction, known as a [Subsidy](#), can be applied at the time of invoice based on the patient's debtor or patient groups.  Taken together, these allow an institution to model a series of complex invoicing scenarios.
```
Comment (jmcameron) [inline docs/en/finance/bookkeeping/patient-invoices.md]: Can you make the "Debtor or patient groups" a link to that document?

File: docs/en/finance/bookkeeping/patient-invoices.md [hunk 2]
```diff
@@ -2,9 +2,11 @@
...
 
-## Invoice Item Prices
-
-If a price list exists, the prices on that list will always override the inventory price. However, the system seeks to apply the lowest price possible for any particular item - if multiple price lists apply to the patient, the items will assume their lowest possible price across all price lists \(need to check that this is the case\).
+## How does BHIMA calculate the price of an item?
 
+At the time of invoicing, BHIMA must compute a price based on three pieces of information:
 
+  1. The institution's price list
+  2. The patient's _patient group_
```
Comment (jmcameron) [inline docs/en/finance/bookkeeping/patient-invoices.md:12]: You seem to be missing the third item?

File: docs/en/finance/bookkeeping/patient-invoices.md [hunk 3]
```diff
@@ -2,9 +2,11 @@
...
 
+If a price list exists, the prices on that list will always override the inventory price. However, the system seeks to apply the lowest price possible for any particular item - if multiple price lists apply to the patient, the items will assume their lowest possible price across all applicable price lists.
```
Comment (jmcameron) [inline docs/en/finance/bookkeeping/patient-invoices.md]: Suggest changing "system" to something more concrete

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Debtor Groups
+
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md:2]: I suggest inserting a definition here of what "debtor" means.

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+## Required Information
+
+Because debtor groups are principally a financial concept, we require certain financial information to create one.  These include:
+
+1. **Name** - the name of the group that will show up in labels and dropdown menus throughout the application.
+2. **Locked** - locks the group to prevent patient assignment and further invoicing of group members.
+3. **Account** - the account used in transactions involving a member of the debtor group.
+4. **Price List** - a price list to apply to the debtor group.
+5. **Max Credit** - _not implemented yet_ [#5068](https://github.com/IMA-WorldHealth/bhima/issues/5068).
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md]: Even though "Max Credit" is not implemented, you should still define it if you list it.  Just leave the "not implemented" at the end.

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+## Conventions / HMOs
+
+A "Convention" is a collective of individuals who are under contract with the medical institution to pay for care of individual members.  It is analogous to an HMO.  Instead of invoicing individual members of the group, the institution will invoice the group for medical care provided to any member of the group.
+
+In BHIMA, conventions are non-cash clients.  This means BHIMA will _block payments at the cash window_ for patients that are in a convention, to prevent double-payment.
+
+## Group Policies
+
+It is often in the best interest of hospitals to charge patients belonging to organisations/HMOs full price to subsidize all the patients who are impoverished and cannot pay.  Conversely, particular debtor groups may a standing relationship with the medical institution to wave administrative fees.
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md]: grammar?  debtor groups may ??? a standing

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+To account for these different scenarios, BHIMA allows administrators to toggle "group policies."  Particular groups can be exempt for subsidies, discounts, or invoicing fees, even if members of the group have subsidies/discounts/fees applied to them from elsewhere.
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md]: That last sentence is confusing.  Are they exempt or not?

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## Subscriptions
+
+Debtor Groups may be subscribed to individual fees or subsidies.  For example, clients that always pay via mobile money may have a processing fee attached to every invoice.  Similarly, church members may receive a subsidy from a religious organisation.
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md]: need a quick parenthetical definition of "mobile money"

**Review by jmcameron — APPROVED** [review]
Good stuff!  
So who is the target audience of these documents?

**Comment by jniles** [comment]
@jmcameron thanks for the review!  I'll get around to fixing these in the next few days.  Thanks for your detailed comments :)

**Comment by jniles** [comment]
I realized I didn't answer your question.  The target audience is my future self, and people trying to poke around a demo application for the first time.

**Comment by jniles** [comment]
@jmcameron this is ready for another review, I think.

File: docs/en/finance/core-concepts/debtor-groups.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+## Required Information
+
+Because debtor groups are principally a financial concept, we require certain financial information to create one.  These include:
+
+1. **Name** - the name of the group that will show up in labels and dropdown menus throughout the application.
+2. **Locked** - locks the group to prevent patient assignment and further invoicing of group members.
+3. **Account** - the account used in transactions involving a member of the debtor group.
+4. **Price List** - a price list to apply to the debtor group.
+5. **Max Credit** - prevents members of the debtor group from being invoice if the debt of the group goes beyond this limit.  _Not implemented yet_ ([#5068](https://github.com/IMA-WorldHealth/bhima/issues/5068)).
```
Comment (jmcameron) [inline docs/en/finance/core-concepts/debtor-groups.md]: Typo:  being invoice**D**

**Review by jmcameron — APPROVED** [review]
Looks great.  I spotted one typo.  Just fix it and release.

**Pull request merged**